src/db.rs

```diff
@@ -637,25 +638,27 @@ impl<T: ThreadMode> DBWithThreadMode<T> {
                     cfhandles.as_mut_ptr(),
                 )),
                 AccessType::Secondary { secondary_path } => {
+                    let secondary_cpath = to_cpath(secondary_path)?;
                     ffi_try!(ffi::rocksdb_open_as_secondary_column_families(
                         opts.inner,
                         cpath.as_ptr() as *const _,
-                        to_cpath(secondary_path)?.as_ptr() as *const _,
+                        secondary_cpath.as_ptr() as *const _,
```

Comment:
I believe this was undefined behavior, because the temporary owned `CString` would have been dropped before the pointer was used in the called function.